iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -1093,5 +1100,16 @@ PortManager::acquireConditionVariableData(const RuntimeName_t& runtimeName) noex
     return m_portPool->addConditionVariableData(runtimeName);
 }
 
+bool PortManager::isInternal(const capro::ServiceDescription& service) const noexcept
+{
+    const capro::ServiceDescription serviceRegistry{
```

Comment:
Please define an `cxx::vector` of all the internal service descriptions (maybe in the `iceoryx_posh_types.hpp`) and iterate over all the entries here to check if they are internal or not.

With such an approach we only have to add new entries to this vector and do not have to adjust the code. Also it would make it more easy to test since we again only have to iterate over the vector to check if they cannot be created.
It also ensures that no internal service is forgotten.

Reply:
Good point, I added the `cxx::vector` to the `PortManager.`